### PR TITLE
Added strlen function

### PR DESCRIPTION
**Pull request by shellhub**
#### Checklist

- [x] Added description of change
- [x] Added file name matches [File name guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#New-File-Name-guidelines)
- [x] Added tests and example, test must pass
- [x] Relevant documentation/comments is changed or added
- [x] PR title follows semantic [commit guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#Commit-Guidelines)
- [x] Search previous suggestions before making a new one, as yours may be a duplicate.
- [x] I acknowledge that all my contributions will be made under the project's license.

Notes: 

<a href="https://gitpod.io/#https://github.com/TheAlgorithms/C/pull/606"></a>

### CODE REVIEW

**Comment by shellhub** [comment]
I created strings folder in this repository and implemented strlen algorithms.

**Comment by shellhub** [comment]
@Panquesito7

**Comment by Panquesito7** [comment]
Are these two files real algorithms?

**Comment by shellhub** [comment]
Yes. I think so.  https://en.m.wikipedia.org/wiki/Algorithm

File: strings/strlen.c [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+ * @file
+ * @brief Program to calculate length of string.
+ *
+ * @author [Du Yuanchao](https://github.com/shellhub)
+ */
+#include <assert.h>
+#include <string.h>
+
+/**
+ * Returns the length of string.
```
Comment (Panquesito7) [inline strings/strlen.c:11]:
```suggestion
 * @brief Returns the length of string
```

File: strings/strlen_recursion.c [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+ * @file
+ * @brief Program to calculate length of string using recursion.
+ *
+ * @author [Du Yuanchao](https://github.com/shellhub)
+ */
+#include <assert.h>
+#include <string.h>
+
+/**
+ * Returns the length of string using recursion.
```
Comment (Panquesito7) [inline strings/strlen_recursion.c:11]:
```suggestion
 * @brief Returns the length of string using recursion
```

**Review by Panquesito7 — APPROVED** [review]
LGTM, good work. 👍 😄

**Comment by shellhub** [comment]
For education. We need this algorithm.

**Pull request merged**